sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1845,12 +1845,12 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
       case SqlBaseParser.PERCENTILE_CONT =>
         sortOrder.direction match {
           case Ascending => PercentileCont(sortOrder.child, percentage)
-          case Descending => PercentileCont(sortOrder.child, Subtract(Literal(1), percentage))
+          case Descending => PercentileCont(sortOrder.child, percentage, true)
```

Comment:
are you sure this won't change query result?

Reply:
The test case output is not changed.